src/game.cpp

```diff
@@ -5488,10 +5490,11 @@ faction *game::faction_by_ident(std::string id)
 {
     for( auto &elem : factions ) {
         if( elem.id == id ) {
+            // TODO: This is really bad, fix this please
```

Comment:
Why? If the backing container has the correct scope this is fine.

Reply:
Could be I'm misremembering, but isn't "elem" here a local variable and thus goes out of scope the instant that we return? This one was definitely marked with a warning of local variable escaping scope.

Reply:
Not if elem is a reference.

Reply:
Strange, not sure why it's reporting as an out of scope error then. Oh well.